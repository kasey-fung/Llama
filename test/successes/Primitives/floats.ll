int main():
	
	float x
	float y
	float z

	/* assignment */
	x = 10.2
	y = 8.0
	print(x) 
	print(y) 


	/* addition */
	z = x + y
	print(z)


	/* subtraction */
	z = x - y
	print(z)
	z = y - x
	print(z)


	/* multiplication */
	z = x * y
	print(z)


	/* division */
	z = x / y
	print(z)


	/* modulus */
	z = x % y
	print(z)
	
	
	/* reassignment */
	x = z
	print(x)


	/* multi-assignment */
	x = z = y
	print(x)
	print(y)
	print(z)

	/* exponentiation */
	x = 10.0
	y = 2.0
	z = x ^ y
	print(z)
