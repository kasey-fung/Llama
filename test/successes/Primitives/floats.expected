10.2
8
18.2
2.2
-2.2
81.6
1.275
2.2
2.2
8
8
8
100